pkg/Library/NodeGraph/Runner.js

```diff
@@ -164,8 +164,8 @@ encodeFullNodeKey({key}, {$meta}) {
 updateNodeInPipeline(node, pipeline) {
   const index = pipeline?.nodes?.findIndex(n => n.key === node.key);
   // TODO (b/245770204): avoid copying objects
-  // pipeline.nodes[index] = node;
-  pipeline.nodes = assign([], pipeline.nodes, {[index]: node});
+  pipeline.nodes[index] = node;
```

Comment:
i have noticed weirdness that occasionally happens when this is done. this is as much as I know...
i think we might want to be cautious and do this change in a separate PR, so it's easily revertable, in case it breaks rapsai (the NodeTree change won't, but Runner will, if...)
or maybe we hold off with this change until i have more time to investigate, WDYT?

Reply:
These concepts should be well understood. 

The code is highly biased towards doing nothing if nothing has happened, we generally want to work in concert with that.